docs/source/llms/rag/notebooks/mlflow-e2e-evaluation.ipynb

```diff
@@ -0,0 +1,918 @@
+    "# To validate the results of a different model, comment out the above line and uncomment the below line:\n",
+    "# eval_results_of_retriever_df_MiniLM = result2.tables[\"eval_results_table\"]\n",
+    "display(eval_results_of_retriever_df_bge)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "application/vnd.databricks.v1+cell": {
+     "cellMetadata": {},
+     "inputWidgets": {},
+     "nuid": "7fecbb62-44ec-4af4-aa5a-7aa79bfa0943",
+     "showTitle": false,
+     "title": ""
+    }
+   },
+   "source": [
+    "### Evaluate different Top K strategy with MLflow"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "application/vnd.databricks.v1+cell": {
+     "cellMetadata": {
+      "byteLimit": 2048000,
+      "rowLimit": 10000
+     },
+     "inputWidgets": {},
+     "nuid": "95a3acd8-170b-4e14-bc51-da977d2b1939",
+     "showTitle": false,
+     "title": ""
+    }
+   },
+   "outputs": [],
+   "source": [
+    "with mlflow.start_run() as run:\n",
+    "    evaluate_results = mlflow.evaluate(\n",
+    "        data=eval_results_of_retriever_df_bge,\n",
+    "        targets=\"source\",\n",
+    "        predictions=\"outputs\",\n",
+    "        evaluators=\"default\",\n",
+    "        extra_metrics=[\n",
+    "            mlflow.metrics.precision_at_k(1),\n",
+    "            mlflow.metrics.precision_at_k(2),\n",
+    "            mlflow.metrics.precision_at_k(3),\n",
+    "            mlflow.metrics.recall_at_k(1),\n",
+    "            mlflow.metrics.recall_at_k(2),\n",
+    "            mlflow.metrics.recall_at_k(3),\n",
+    "            mlflow.metrics.ndcg_at_k(1),\n",
+    "            mlflow.metrics.ndcg_at_k(2),\n",
+    "            mlflow.metrics.ndcg_at_k(3),\n",
+    "        ],\n",
+    "    )\n",
+    "\n",
+    "display(evaluate_results.tables[\"eval_results_table\"])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "application/vnd.databricks.v1+cell": {
+     "cellMetadata": {},
+     "inputWidgets": {},
+     "nuid": "74a52bda-1ea7-4f50-abac-e36d78e1b96b",
+     "showTitle": false,
+     "title": ""
+    }
+   },
+   "source": [
+    "### Evaluate the Chunking Strategy with MLflow"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "application/vnd.databricks.v1+cell": {
+     "cellMetadata": {
+      "byteLimit": 2048000,
+      "rowLimit": 10000
+     },
+     "inputWidgets": {},
+     "nuid": "d2c6bff7-d988-4f09-ac10-6c9ea14b9242",
+     "showTitle": false,
+     "title": ""
+    }
+   },
+   "outputs": [],
+   "source": [
+    "def evaluate_chunk_size(chunk_size):\n",
+    "    list_of_documents = loader.load()\n",
+    "    text_splitter = CharacterTextSplitter(chunk_size=chunk_size, chunk_overlap=0)\n",
+    "    docs = text_splitter.split_documents(list_of_documents)\n",
+    "    embedding_function = SentenceTransformerEmbeddings(model_name=\"all-MiniLM-L6-v2\")\n",
```

Comment:
```suggestion
    "    embedding_function = DatabricksEmbeddings(endpoint=\"databricks-bge-large-en\")\n",
```

For consistency

Reply:
Good catch

Reply:
I left a comment in the cell above (going to push soon) that explains why this is different from above so that users can see the difference in the output results between the different embeddings models.